src/test/browserutils.ts

```diff
@@ -48,14 +48,13 @@ export const navigate = async(target: puppeteer.Page, path: string) => {
 };
 
 export const finishRender = async(target: puppeteer.Page) => {
-  // wait for any new async hooks to register...
-  await new Promise((resolve) => setTimeout(resolve, 1000));
+  // HACK
+  // - there is no convenient way to tell if chrome is finished rendering fonts and things.
+  // - its hard to tell when simulations are done loading.
```

Comment:
can the `readyState` be polled? (Not sure it is worth adding in the complexity into production code but maybe tests can use it? Something like https://stackoverflow.com/a/36155560

Similarly for fonts, I think MathJax creates a `<span style="font-face: FancyFont">W</span>` and polls to check the width to see if the font has loaded. Not sure it is worth adding